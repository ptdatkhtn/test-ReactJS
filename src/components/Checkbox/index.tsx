import Checkbox from './styles'

export default Checkbox
