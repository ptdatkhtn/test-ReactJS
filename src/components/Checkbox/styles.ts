import { Checkbox } from 'antd'
import styled from 'styled-components'

export default styled(Checkbox)``
