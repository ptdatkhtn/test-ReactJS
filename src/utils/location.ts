export const degreeToRadian = (degree : number): number => {
  return degree * Math.PI / 180
}

export const calculateDistanceInKm = (lat1: number, lon1: number, lat2: number, lon2: number): number => {
  const earthRadiusKm = 6371

  // have to convert degree unit to radian unit before the calculation
  const dLat = degreeToRadian(lat2 - lat1)
  const dLon = degreeToRadian(lon2 - lon1)

  lat1 = degreeToRadian(lat1)
  lat2 = degreeToRadian(lat2)

  const a = Math.sin(dLat / 2) * Math.sin(dLat / 2) + Math.sin(dLon / 2) * Math.sin(dLon / 2) * Math.cos(lat1) * Math.cos(lat2)
  const c = 2 * Math.atan2(Math.sqrt(a), Math.sqrt(1 - a))
  return earthRadiusKm * c
}