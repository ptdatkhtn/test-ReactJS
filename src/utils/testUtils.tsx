import { render as rtlRender } from '@testing-library/react'
import React from 'react'
import { ThemeProvider } from 'styled-components'
import theme from 'themes'

const render = (ui: JSX.Element, { ...renderOptions } = {}) => {
  const Wrapper = ({ children }: { children: React.ReactNode }) => {
    return (
      <ThemeProvider theme={theme}>
        {children}
      </ThemeProvider>
    )
  }
  return rtlRender(ui, { wrapper: Wrapper as React.FC, ...renderOptions })
}

const fixMatchMedia = () => {
  window.matchMedia = window.matchMedia || function() {
    return {
        matches: false,
        addListener: function() {},
        removeListener: function() {}
    }
  }
}

export * from '@testing-library/react'
// override render method from @testing-library/react
export { render, fixMatchMedia }
