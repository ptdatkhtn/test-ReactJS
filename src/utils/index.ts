import { calculateDistanceInKm } from './location'

export { calculateDistanceInKm }