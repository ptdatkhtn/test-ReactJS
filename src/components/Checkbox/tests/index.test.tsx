import React from 'react'
import 'jest-styled-components'
import Checkbox from '..'
import { render } from 'utils/testUtils'

describe('Checkbox', () => {
  it('should match snapshot', () => {
    const { container } = render(
      <>
        <Checkbox>Label</Checkbox>
        <Checkbox disabled>Label</Checkbox>
        <Checkbox checked>Label</Checkbox>
        <Checkbox checked disabled>
          Label
        </Checkbox>
      </>
    )
    expect(container).toMatchSnapshot()
  })
})