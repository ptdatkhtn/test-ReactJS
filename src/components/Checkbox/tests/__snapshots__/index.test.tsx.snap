// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should match snapshot 1`] = `
<div>
  <label
    class="ant-checkbox-wrapper "
  >
    <span
      class="ant-checkbox"
    >
      <input
        class="ant-checkbox-input"
        type="checkbox"
        value=""
      />
      <span
        class="ant-checkbox-inner"
      />
    </span>
    <span>
      Label
    </span>
  </label>
  <label
    class="ant-checkbox-wrapper ant-checkbox-wrapper-disabled "
  >
    <span
      class="ant-checkbox ant-checkbox-disabled"
    >
      <input
        class="ant-checkbox-input"
        disabled=""
        type="checkbox"
        value=""
      />
      <span
        class="ant-checkbox-inner"
      />
    </span>
    <span>
      Label
    </span>
  </label>
  <label
    class="ant-checkbox-wrapper ant-checkbox-wrapper-checked "
  >
    <span
      class="ant-checkbox ant-checkbox-checked"
    >
      <input
        checked=""
        class="ant-checkbox-input"
        type="checkbox"
        value=""
      />
      <span
        class="ant-checkbox-inner"
      />
    </span>
    <span>
      Label
    </span>
  </label>
  <label
    class="ant-checkbox-wrapper ant-checkbox-wrapper-checked ant-checkbox-wrapper-disabled "
  >
    <span
      class="ant-checkbox ant-checkbox-checked ant-checkbox-disabled"
    >
      <input
        checked=""
        class="ant-checkbox-input"
        disabled=""
        type="checkbox"
        value=""
      />
      <span
        class="ant-checkbox-inner"
      />
    </span>
    <span>
      Label
    </span>
  </label>
</div>
`;
